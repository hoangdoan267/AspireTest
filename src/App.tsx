/* eslint-disable react-native/no-inline-styles */
import * as React from 'react';
import {View, StatusBar, Image} from 'react-native';
import {NavigationContainer} from '@react-navigation/native';
import {createBottomTabNavigator} from '@react-navigation/bottom-tabs';
import {createStackNavigator} from '@react-navigation/stack';
import {SafeAreaProvider} from 'react-native-safe-area-context';

import {DebitCardScreen, SpendingLimitScreen} from './screens';
import {PRIMARY_COLOR, COLOR_BLUE} from './helper/Constants';
import {Provider} from 'react-redux';
import {store} from './store/store';
import {TabBar} from './component';

const BlankScreen = () => {
  return (
    <View style={{flex: 1, justifyContent: 'center', alignItems: 'center'}}>
      <Image
        source={require('./assets/images/logo.png')}
        style={{height: 64, width: 64}}
      />
    </View>
  );
};

const HomeTabs = () => {
  return (
    <Tab.Navigator
      initialRouteName="Debit Card"
      tabBar={(props) => <TabBar {...props} />}
      tabBarOptions={{
        activeTintColor: PRIMARY_COLOR,
        inactiveTintColor: 'grey',
      }}>
      <Tab.Screen name="Home" component={BlankScreen} />
      <Tab.Screen name="Debit Card" component={DebitCardScreen} />
      <Tab.Screen name="Payment" component={BlankScreen} />
      <Tab.Screen name="Credit" component={BlankScreen} />
      <Tab.Screen name="Profile" component={BlankScreen} />
    </Tab.Navigator>
  );
};

const Tab = createBottomTabNavigator();
const Stack = createStackNavigator();

const App = () => {
  return (
    <Provider store={store}>
      <SafeAreaProvider>
        <NavigationContainer>
          <StatusBar barStyle="light-content" backgroundColor={COLOR_BLUE} />
          <Stack.Navigator
            initialRouteName="Home"
            screenOptions={{
              headerShown: false,
            }}>
            <Stack.Screen name="Home" component={HomeTabs} />
            <Stack.Screen
              name="SpendingLimit"
              component={SpendingLimitScreen}
            />
          </Stack.Navigator>
        </NavigationContainer>
      </SafeAreaProvider>
    </Provider>
  );
};

export default App;
