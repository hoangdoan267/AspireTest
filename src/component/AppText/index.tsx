import React from 'react';

import {Text} from 'react-native';

export const AppText = (props: any) => {
  return <Text {...props}>{props.children}</Text>;
};
