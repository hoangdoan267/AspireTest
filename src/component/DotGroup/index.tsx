import React from 'react';
import {View, Text} from 'react-native';
import {styles} from './styles';

export const DotGroup = () => {
  return (
    <View style={styles.dotGroup}>
      <View style={styles.dot} />
      <View style={styles.dot} />
      <View style={styles.dot} />
      <View style={styles.dot} />
    </View>
  );
};
