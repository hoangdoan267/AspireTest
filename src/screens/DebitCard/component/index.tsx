import {LimitBar} from './LimitBar';

export {LimitBar};
