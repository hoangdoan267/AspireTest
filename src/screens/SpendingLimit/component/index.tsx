import {LimitAmount} from './LimitAmount';

export {LimitAmount};
