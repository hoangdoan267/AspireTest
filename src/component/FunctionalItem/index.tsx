import React from 'react';
import {View, Switch} from 'react-native';
import {styles} from './styles';
import {AppText} from '../AppText';
import {TouchableOpacity} from 'react-native-gesture-handler';

interface Props {
  icon: JSX.Element;
  name?: string;
  content?: string;
  haveSwitch?: boolean;
  switchValue?: boolean;
  onChange?: (value: boolean) => void;
  onPress?: () => void;
}

export const FunctionalItem = (props: Props) => {
  const {
    icon,
    name,
    content,
    haveSwitch,
    onChange,
    switchValue,
    onPress,
  } = props;

  const onChangeSwitch = (value: boolean) => {
    onChange && onChange(value);
  };

  return (
    <View style={styles.itemContainer}>
      <TouchableOpacity
        style={styles.contentContainer}
        activeOpacity={0.9}
        onPress={onPress}>
        {icon}
        <View>
          <AppText style={styles.nameStyle}>{name}</AppText>
          <AppText style={styles.contentStyle}>{content}</AppText>
        </View>
      </TouchableOpacity>
      {haveSwitch && (
        <Switch
          onValueChange={onChangeSwitch}
          value={switchValue}
          style={{transform: [{scaleX: 0.7}, {scaleY: 0.7}]}}
        />
      )}
    </View>
  );
};
