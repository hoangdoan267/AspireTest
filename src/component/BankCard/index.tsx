import React, {useState} from 'react';
import {View, Text, Image} from 'react-native';
import {styles} from './styles';
import {TouchableOpacity} from 'react-native-gesture-handler';
import {AppText} from '../AppText';
import {DotGroup} from '../DotGroup';

interface Props {
  cardInformation: {
    name: string;
    cardNumber: string;
    expired: string;
    cvv: string;
  };
}

export const BankCard = (props: Props) => {
  const {name, cardNumber, expired, cvv} = props.cardInformation;

  const [isHide, setIsHide] = useState<boolean>(true);

  const renderCardNumber = () => {
    if (!isHide) {
      return (
        <View style={styles.infoGroup}>
          <AppText style={styles.cardNumber}>
            {`${cardNumber.substr(0, 4)}   ${cardNumber.substr(
              4,
              4,
            )}   ${cardNumber.substr(8, 4)}  ${cardNumber.substr(12, 4)}`}
          </AppText>
        </View>
      );
    }
    return (
      <View style={styles.infoGroup}>
        <DotGroup />
        <DotGroup />
        <DotGroup />
        <AppText style={styles.cardNumber}>{cardNumber.substr(12, 4)}</AppText>
      </View>
    );
  };

  const cvvSection = () => {
    if (isHide) {
      return (
        <View style={styles.cvvSection}>
          <AppText style={styles.cvvText}>CVV:</AppText>
          <AppText style={styles.cvvHideText}>***</AppText>
        </View>
      );
    }
    return (
      <View style={[styles.cvvSection]}>
        <AppText style={styles.infoText}>CVV: {cvv}</AppText>
      </View>
    );
  };

  return (
    <>
      <View style={styles.hideContainer}>
        <TouchableOpacity
          style={styles.hideButton}
          activeOpacity={1}
          onPress={() => setIsHide(!isHide)}>
          {isHide ? (
            <Image
              source={require('../../assets/images/show.png')}
              style={styles.hideIcon}
            />
          ) : (
            <Image
              source={require('../../assets/images/hide.png')}
              style={styles.hideIcon}
            />
          )}
          <AppText style={styles.hideText}>
            {isHide ? 'Show card number' : 'Hide card number'}
          </AppText>
        </TouchableOpacity>
      </View>

      <View style={styles.cardContainer}>
        <View style={styles.logoArea}>
          <Image
            source={require('../../assets/images/white_logo.png')}
            style={styles.logoStyle}
          />
        </View>
        <AppText style={styles.cardName}>{name}</AppText>
        {renderCardNumber()}
        <View style={[styles.infoGroup, {paddingTop: !isHide ? 8 : 0}]}>
          <AppText style={styles.infoText}>Thru: {expired}</AppText>
          {cvvSection()}
        </View>

        <View style={styles.visaLogoArea}>
          <Image
            source={require('../../assets/images/visa.png')}
            style={styles.visaLogoStyle}
          />
        </View>
      </View>
    </>
  );
};
