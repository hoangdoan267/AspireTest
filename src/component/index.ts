import {BankCard} from './BankCard';
import {AppText} from './AppText';
import {FunctionalItem} from './FunctionalItem';
import {Header} from './Header';
import {DotGroup} from './DotGroup';
import {MoneyTag} from './MoneyTag';
import {TabBar} from './TabBar';

export {BankCard, AppText, FunctionalItem, Header, DotGroup, MoneyTag, TabBar};
