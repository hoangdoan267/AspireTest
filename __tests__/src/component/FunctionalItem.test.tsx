import React from 'react';
import renderer from 'react-test-renderer';
import {FunctionalItem} from '../../../src/component';
import {Image} from 'react-native';

import {render, fireEvent} from '@testing-library/react-native';

describe('render FunctionalItem', () => {
  test('render correctly', () => {
    const snapshot = renderer
      .create(
        <FunctionalItem
          name="Hello World"
          content="Lorem ipsum"
          icon={
            <Image source={require('../../../src/assets/images/limit.png')} />
          }
        />,
      )
      .toJSON();

    expect(snapshot).toMatchSnapshot();
  });

  test('render with switch', () => {
    let onActionMock = jest.fn();

    const snapshot = renderer
      .create(
        <FunctionalItem
          name="Hello World"
          content="Lorem ipsum"
          haveSwitch
          icon={
            <Image source={require('../../../src/assets/images/limit.png')} />
          }
          onChange={onActionMock}
        />,
      )
      .toJSON();

    expect(snapshot).toMatchSnapshot();
  });
});
