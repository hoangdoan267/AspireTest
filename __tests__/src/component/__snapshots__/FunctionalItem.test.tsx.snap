// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render FunctionalItem render correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginTop": 32,
      "width": "100%",
    }
  }
>
  <RNGestureHandlerButton
    collapsable={false}
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    rippleColor={0}
  >
    <View
      accessible={true}
      style={
        Object {
          "flexDirection": "row",
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/images/limit.png",
          }
        }
      />
      <View>
        <Text
          style={
            Object {
              "fontSize": 14,
              "marginBottom": 4,
            }
          }
        >
          Hello World
        </Text>
        <Text
          style={
            Object {
              "fontSize": 13,
              "opacity": 0.4,
              "width": 240,
            }
          }
        >
          Lorem ipsum
        </Text>
      </View>
    </View>
  </RNGestureHandlerButton>
</View>
`;

exports[`render FunctionalItem render with switch 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginTop": 32,
      "width": "100%",
    }
  }
>
  <RNGestureHandlerButton
    collapsable={false}
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    rippleColor={0}
  >
    <View
      accessible={true}
      style={
        Object {
          "flexDirection": "row",
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/images/limit.png",
          }
        }
      />
      <View>
        <Text
          style={
            Object {
              "fontSize": 14,
              "marginBottom": 4,
            }
          }
        >
          Hello World
        </Text>
        <Text
          style={
            Object {
              "fontSize": 13,
              "opacity": 0.4,
              "width": 240,
            }
          }
        >
          Lorem ipsum
        </Text>
      </View>
    </View>
  </RNGestureHandlerButton>
  <RCTSwitch
    accessibilityRole="switch"
    onChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "height": 31,
          "width": 51,
        },
        Object {
          "transform": Array [
            Object {
              "scaleX": 0.7,
            },
            Object {
              "scaleY": 0.7,
            },
          ],
        },
      ]
    }
    testID="switch-functional"
    value={false}
  />
</View>
`;
